File: Python Fundamentals/PF Exercises/4) Functions - Exercises/ex4_odd_and_even_sum.py
def odd_even(digits: str):
    even_sum = 0
    odd_sum = 0
    for digit in digits:
        if int(digit) % 2 == 0:
            even_sum += int(digit)
        else:
            odd_sum += int(digit)
    return even_sum, odd_sum


numbers = input()
sum_even_digits, sum_odd_digits = odd_even(numbers)
print(f"Odd sum = {sum_odd_digits}, Even sum = {sum_even_digits}")

File: JS Front-End/Exercises/3) Functions and Statements - Exercises/ex4-odd-and-even-sum.js
function solve (number) {

    const digits = number.toString().split('').map(Number)
    const evenSum = digits.filter(d => d % 2 === 0).reduce((sum, d) => sum + d, 0)
    const oddSum = digits.filter(d => d % 2 !== 0).reduce((sum, d) => sum + d, 0)

    console.log(`Odd sum = ${oddSum}, Even sum = ${evenSum}`)

    // const numAsString = String(number);
    // let evenSum = 0;
    // let oddSum = 0;

    // for (let i = 0; i < numAsString.length; i++) {
    //     const num = Number(numAsString[i]);

    //     if (num % 2 === 0) {
    //         evenSum += num
    //     } else {
    //         oddSum += num
    //     }
    // }

    // console.log(`Odd sum = ${oddSum}, Even sum = ${evenSum}`)
}

